### DISCUSSION TITLE
Inconsistent behaviour of updating workflow files with `GITHUB_TOKEN`

### DISCUSSION THREAD

**Discussion by algomaster99** [community/community#25222] [Actions]
<p>Hey! I have created a <a href="https://github.com/SpoonLabs/sorald/blob/08ec1ee2b0fac18981a4a7c83b938685a2a0aded/.github/workflows/combine-prs.yml" rel="noopener nofollow ugc">workflow</a> file that is supposed to <a href="https://github.com/SpoonLabs/sorald/blob/08ec1ee2b0fac18981a4a7c83b938685a2a0aded/.github/workflows/combine-prs.yml#L136" rel="noopener nofollow ugc">combine branches</a> into one and <a href="https://github.com/SpoonLabs/sorald/blob/08ec1ee2b0fac18981a4a7c83b938685a2a0aded/.github/workflows/combine-prs.yml#L137" rel="noopener nofollow ugc">push the newly created branch</a> to the repository. This workflow is triggered manually and uses the <code>GITHUB_TOKEN</code> default permissions.</p>
<p>I have noticed inconsistent behaviour in how it works. When the newly-created branch contains commits with edits to a workflow file, it is supposed to fail because <code>GITHUB_TOKEN</code> does not have permissions to edit workflow file by default. However, you can see that it <a href="https://github.com/SpoonLabs/sorald/runs/5853219507?check_suite_focus=true#step:4:37" rel="noopener nofollow ugc">passes</a> (<code>git push</code> is performed successfully) here, but fails <a href="https://github.com/SpoonLabs/sorald/runs/5854403799?check_suite_focus=true#step:4:36" rel="noopener nofollow ugc">here</a>.</p>
<p>Is this a bug or am I doing something wrong?</p>

**Comment by jsoref** [comment]
<p>I believe the distinction is in one case you’re <a href="https://github.com/SpoonLabs/sorald/runs/5853219507?check_suite_focus=true#step:4:37">creating a new branch</a> and in the other <a href="https://github.com/SpoonLabs/sorald/runs/5854403799?check_suite_focus=true#step:4:36">you’re trying to update an existing branch</a>.</p>
<p>The short of it is if you want to update an existing branch, you’ll need to bring your own token.</p>
<p>You could use:</p><aside class="onebox allowlistedgeneric" data-onebox-src="https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token">
  <header class="source">
      <img src="https://docs.github.com/assets/cb-803/images/site/favicon.svg" class="site-icon" width="16" height="16">

      <a href="https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token" target="_blank" rel="noopener">GitHub Docs</a>
  </header>

  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/2119212/181089669-befb16b5-8306-49f7-9744-d081fe1fb34b.png" class="thumbnail onebox-avatar" width="500" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_500x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_750x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_1000x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_10x10.png">

<h3><a href="https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token" target="_blank" rel="noopener">Creating a personal access token - GitHub Docs</a></h3>

  <p>You should create a personal access token to use in place of a password with the command line or with the API.</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>
<p>
or</p><aside class="onebox allowlistedgeneric" data-onebox-src="https://docs.github.com/en/developers/overview/managing-deploy-keys">
  <header class="source">
      <img src="https://docs.github.com/assets/cb-803/images/site/favicon.svg" class="site-icon" width="16" height="16">

      <a href="https://docs.github.com/en/developers/overview/managing-deploy-keys" target="_blank" rel="noopener">GitHub Docs</a>
  </header>

  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/2119212/181089671-82830292-f313-4440-82d1-f4568603ee37.png" class="thumbnail onebox-avatar" width="500" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_500x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_750x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_1000x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_10x10.png">

<h3><a href="https://docs.github.com/en/developers/overview/managing-deploy-keys" target="_blank" rel="noopener">Managing deploy keys - GitHub Docs</a></h3>

  <p>Learn different ways to manage SSH keys on your servers when you automate deployment scripts and which way is best for you.</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>Note: because you’re using the GitHub Actions app to create the branch, workflows won’t run on that push, and thus there are no direct possible side-effects from creating the branch. (You could have a webhook that foolishly does something in response, but if you do, that’s entirely on you.)</p>
<p>Fwiw, here’s how I’ve addressed similar problems:</p><aside class="onebox allowlistedgeneric" data-onebox-src="https://github.com/check-spelling/check-spelling/wiki/Feature:-Update-with-deploy-key">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">

      <a href="https://github.com/check-spelling/check-spelling/wiki/Feature:-Update-with-deploy-key" target="_blank" rel="noopener">GitHub</a>
  </header>

  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/345;"><img src="https://user-images.githubusercontent.com/2119212/181089672-98019b61-1b98-4bf0-b19d-98d70c8b6d9c.png" class="thumbnail" width="690" height="345" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/4/9/49c695e11920dccafce1cbea1a2b8887f8f4c6e5_2_690x345.png, https://aws1.discourse-cdn.com/github/optimized/3X/4/9/49c695e11920dccafce1cbea1a2b8887f8f4c6e5_2_1035x517.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/4/9/49c695e11920dccafce1cbea1a2b8887f8f4c6e5.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/4/9/49c695e11920dccafce1cbea1a2b8887f8f4c6e5_2_10x10.png"></div>

<h3><a href="https://github.com/check-spelling/check-spelling/wiki/Feature:-Update-with-deploy-key" target="_blank" rel="noopener">Feature: Update with deploy key · check-spelling/check-spelling Wiki</a></h3>

  <p>Spelling checker action. Contribute to check-spelling/check-spelling development by creating an account on GitHub.</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>I haven’t specifically tried to generally update the workflow itself, but it’s something I’m thinking about.</p>

**Comment by algomaster99** [comment]
<blockquote>
<p>I believe the distinction is in one case you’re creating a new branch and in the other you’re trying to update an existing branch.</p>
</blockquote>
<p>I tried triggering the workflow again after giving a new name to the combined branch (<code>renovate-dependencies-combined</code>), but it still failed. See the failed workflow <a href="https://github.com/SpoonLabs/sorald/runs/5869629363?check_suite_focus=true" rel="noopener nofollow ugc">here</a>.</p>
<blockquote>
<p>because you’re using the GitHub Actions app to create the branch, workflows won’t run on that push</p>
</blockquote>
<p>I am aware of that. I also found some workarounds that are listed <a href="https://github.com/peter-evans/create-pull-request/blob/f1a7646cead32c950d90344a4fb5d4e926972a8f/docs/concepts-guidelines.md#triggering-further-workflow-runs" rel="noopener nofollow ugc">here</a>. Thanks for you suggestion too!</p>

**Comment by jsoref** [comment, marked as answer]
<p>Ah, sorry, I assumed that your things were equivalent content. That was a mistake on my part.</p>
<p>But, at the end of the day, the github-actions bot doesn’t have <code>workflows</code> permission and you can’t ask for it to have it:</p>
<aside class="onebox allowlistedgeneric" data-onebox-src="https://github.blog/changelog/2021-04-20-github-actions-control-permissions-for-github_token/#setting-permissions-in-the-workflow">
  <header class="source">
      <img src="https://user-images.githubusercontent.com/2119212/181089674-203211cf-5c98-4d99-ad5f-66a214217e53.png" class="site-icon" width="32" height="32">

      <a href="https://github.blog/changelog/2021-04-20-github-actions-control-permissions-for-github_token/#setting-permissions-in-the-workflow" target="_blank" rel="noopener">The GitHub Blog</a>
  </header>

  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/195;"><img src="https://user-images.githubusercontent.com/2119212/181089676-6899305d-93d9-4c0e-a0d9-bf39311fe0a0.png" class="thumbnail" width="690" height="195" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/0/0/004437e6fcfdb9237c8b8c5c22bb00e62e034ba2_2_690x195.png, https://aws1.discourse-cdn.com/github/optimized/3X/0/0/004437e6fcfdb9237c8b8c5c22bb00e62e034ba2_2_1035x292.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/0/0/004437e6fcfdb9237c8b8c5c22bb00e62e034ba2_2_1380x390.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/0/0/004437e6fcfdb9237c8b8c5c22bb00e62e034ba2_2_10x10.png"></div>

<h3><a href="https://github.blog/changelog/2021-04-20-github-actions-control-permissions-for-github_token/#setting-permissions-in-the-workflow" target="_blank" rel="noopener">GitHub Actions: Control permissions for GITHUB_TOKEN | GitHub Changelog</a></h3>

  <p>GitHub Actions: Control permissions for GITHUB_TOKEN</p>

  <p>
    <span class="label1">Est. reading time: 1 minute</span>
  </p>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>So, you’ll need to use an alternate credential.</p>

**Comment by algomaster99** [comment]
<p><a class="mention" href="/jsoref">@jsoref</a> Thank you for the prompt response. I will setup an SSH key to perform deployments in that case.</p>
